benchmarks/modules/utils.py

```diff
@@ -267,11 +271,38 @@ def setup_spack_environment(self):
             f'(cd {cp_dir}; find . \( -name "spack.yaml" -o -name "compilers.yaml" -o -name "packages.yaml" \) -print0 | xargs -0 tar cf - | tar -C {dest} -xvf -)',
             f'spack -e {self.build_system.environment} config add "config:install_tree:root:{env_dir}/opt"',
         ]
-
+        # else f"{d[key].value}" if isinstance(d[key].value, bool) for boolean with "True" instead of True
+        cmd_spack_variants = 'from spack import environment; d = environment.active_environment().spec_lists["specs"].specs[0].variants.dict; keys = list(d.keys());values = {key: ([d[key].value[i] for i in range(len(d[key].value))] if isinstance(d[key].value, tuple) else d[key].value) for key in keys};print(values)'
```

Comment:
Is this the whole variant string broken down into keys automatically or as one string?

Reply:
Actually.... If this contains the compiler name and version as well, do we still need the separate variables and commands to fish them out? Can we let the post-processing unpack it into separate columns instead?

Reply:
I thought it would be better to keep variants and compiler information separate since compiler name and version is _always_ available but some applications won't even have a variant at all maybe and spack_spec as a string is available in the columns as a whole anyway.